test/server/cards/02_SHD/upgrades/RichReward.spec.ts

```diff
@@ -0,0 +1,95 @@
+describe('Rich Reward', function() {
+    integration(function(contextRef) {
+        describe('Rich Reward\'s Bounty ability', function() {
+            it('should give Experience token up to two units', function () {
+                contextRef.setupTest({
+                    phase: 'action',
+                    player1: {
+                        groundArena: [
+                            { card: 'battlefield-marine', upgrades: ['rich-reward'] },
+                            { card: 'atst' },
+                            { card: 'yoda#old-master' }
+                        ]
+                    },
+                    player2: {
+                        groundArena: ['wampa'],
+                        spaceArena: ['concord-dawn-interceptors']
+                    }
+                });
+
+                const { context } = contextRef;
+
+                context.player1.clickCard(context.battlefieldMarine);
+                context.player1.clickCard(context.wampa);
+
+                expect(context.player2).toBeAbleToSelectExactly([context.wampa, context.atst, context.yoda, context.concordDawnInterceptors]);
+                expect(context.player2).toHavePassAbilityButton();
+                expect(context.player2).toHavePrompt('Select 2 cards');
+                context.player2.clickCard(context.wampa);
+                context.player2.clickCard(context.concordDawnInterceptors);
```

Comment:
We should check that no more than 2 cards can be selected. The way to do that is to click a third card using `clickCardNonChecking()` instead of `clickCard()`, and then confirm that the third clicked unit does not get an experience

Reply:
I've added a test for that case, please let me know if I'm missing something

Reply:
Oh we don't need a whole extra case for that, especially since we're trying to consolidate to as few individual cases as possible. Apologies for my being unclear, but do you mind just adding the click here and removing the other test? Unless there's something else that test is covering that this test isn't

Reply:
my bad, I should simplify the unit test. Done